client/src/containers/CampaignAdmin/CampaignAdmin.js

```diff
@@ -206,13 +218,49 @@ class CampaignAdmin extends SilverStripeComponent {
       itemListViewEndpoint: this.props.sectionConfig.itemListViewEndpoint,
       publishApi: this.publishApi,
       handleBackButtonClick: this.handleBackButtonClick.bind(this),
+      onRemoveCampaignItem: this.handleRemoveCampaignItem,
     };
 
     return (
       <CampaignAdminList {...props} />
     );
   }
 
+  handleRemoveCampaignItem(campaignId, itemId) {
+    const fallbackMsg = `Are you sure you want to remove this item?
+
+By removing this item all linked items will be removed unless used elsewhere.`;
+    const msg = i18n._t('CampaignAdmin.REMOVE_ITEM_MESSAGE', fallbackMsg);
+    // eslint-disable-next-line no-alert
+    const confirmed = window.confirm(msg);
+
+    if (!confirmed) {
+      return false;
+    }
+
+    this.props.campaignActions.removeCampaignItem(
+      this.removeCampaignItemApi,
+      campaignId,
+      itemId
+    )
+      .then(() => {
+        const endpoint = this.props.sectionConfig.readCampaignsEndpoint;
+        const fetchURL = endpoint.url;
+        this.props.recordActions.fetchRecords(
+          this.props.sectionConfig.treeClass,
+          endpoint.method,
+          fetchURL
+        )
+          .then(() => {
+            this.props.campaignActions.selectChangeSetItem(null);
+            // Workaround to hide more actions popover
+            window.document.body.click();
+          });
+      });
+
+    return true;
```

Comment:
So the reason to return the promise, rather than this empty `true` is for unit testing...

I would like to see a unit test covering the functions that need to be called in this chain.